tensorflow/python/keras/saving/save.py

```diff
@@ -92,12 +90,42 @@ def save_model(model,
   """
   from tensorflow.python.keras.engine import sequential  # pylint: disable=g-import-not-at-top
 
-  default_format = 'tf' if tf2.enabled() else 'h5'
-  save_format = save_format or default_format
-
-  if (save_format == 'h5' or
-      (h5py is not None and isinstance(filepath, h5py.File)) or
-      os.path.splitext(filepath)[1] in _HDF5_EXTENSIONS):
+  if type(filepath) != str and not isinstance(filepath, h5py.File):
```

Comment:
Perhaps refactor out this part of the code to a separate function that is called by both `save_weights()` and `save()`. 

Also, it would be useful to have unit tests to make sure that the save format returned by the function is correct